### PR TITLE
simplifies packet/shred sanity checks

### PR DESCRIPTION
**Pull request by behzadnouri**
#### Problem
Some sanity checks are duplicate and so redundant.

#### Summary of Changes
deduped sanity checks.

### CODE REVIEW

File: ledger/src/shred.rs [hunk 1]
```diff
@@ -532,9 +532,7 @@ pub mod layout {
 
     pub fn get_shred(packet: &Packet) -> Option<&[u8]> {
         let size = get_shred_size(packet)?;
-        let shred = packet.data(..size)?;
-        // Should at least have a signature.
-        (size >= SIZE_OF_SIGNATURE).then(|| shred)
```
Comment (behzadnouri) [inline ledger/src/shred.rs:537]: all subsequent `shred::layout` method do verify size before offsetting into the buffer.

File: ledger/src/shred.rs [hunk 2]
```diff
@@ -682,21 +680,29 @@ impl TryFrom<u8> for ShredVariant {
...
     let shred = match layout::get_shred(packet) {
         None => {
             stats.index_overrun += 1;
             return true;
         }
         Some(shred) => shred,
     };
-    if OFFSET_OF_SHRED_INDEX + SIZE_OF_SHRED_INDEX > shred.len() {
-        stats.index_overrun += 1;
-        return true;
```
Comment (behzadnouri) [inline ledger/src/shred.rs:699]: already verified by `layout::get_version` which access a higher offset.

File: ledger/src/shred.rs [hunk 3]
```diff
@@ -682,21 +680,29 @@ impl TryFrom<u8> for ShredVariant {
 pub fn should_discard_shred(
     packet: &Packet,
     root: Slot,
+    max_slot: Slot, // Max slot to ingest shreds for.
```
Comment (steviez) [inline ledger/src/shred.rs:683]: Given that the lower/upper bound check are now separate, what do you think about moving this comment to above function declaration, and also mention `root`, maybe something like:
```
// Accept shreds in the slot range [root + 1, max_slot]
```

**Review by steviez — APPROVED** [review]
Just the one suggestion on comment, otherwise, LGTM!

File: ledger/src/shred.rs (hunk 3 above)
Comment (behzadnouri) [reply ledger/src/shred.rs]: done.

**Review by steviez — APPROVED** [review]

**Pull request merged**